modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/title_editor/LocalizedInput.es.js

```diff
@@ -0,0 +1,144 @@
+import getCN from 'classnames';
+import LocalizedDropdown from './LocalizedDropdown.es';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class LocalizedInput extends React.Component {
+	static propTypes = {
+		availableLanguages: PropTypes.object.isRequired,
+		initialLanguageId: PropTypes.string.isRequired,
+		initialOpen: PropTypes.bool,
+		initialValues: PropTypes.object,
+		onChange: PropTypes.func,
+		readOnly: PropTypes.bool
+	}
+	static defaultProps = {
+		initialOpen: false,
+		initialValues: {},
+		onChange: () => {},
+		readOnly: false
+	}
+
+	state = {
+		availableLanguages: Object.entries(this.props.availableLanguages).map(
```

Comment:
Since you are accessing `this.props` here, I think you need to be setting this initial state within the `constructor`

Reply:
Yeh, I thought so too. But it is the same thing. For consistency we can go either way, because I use to moved such state definitions to `constructor`, but it turns out it's not necessary.

Reply:
Ah I see, since thats the case lets just pick one or the other and then try to stick with that pattern. I am fine with either approach so long as we keep it consistent.

Reply:
Ok, I'm going for the `constructor` as most of the times I cannot define the `state` from `props` in a one-liner.